cmd/item.go

```diff
@@ -0,0 +1,134 @@
+	}
+}
+
+func addBoolFlags(command *cobra.Command, metadata map[string]FlagMetadata[bool]) {
+	for _, value := range metadata {
+		if value.IsInheritable {
+			var inheritFlagName = value.Name + "-inherit"
+			command.Flags().Bool(value.Name, value.DefaultValue, value.Usage)
+			command.Flags().Bool(
+				inheritFlagName,
+				false,
+				fmt.Sprintf(inheritedUsageFormat, value.Name),
+			)
+		} else {
+			command.Flags().Bool(value.Name, value.DefaultValue, value.Usage)
+		}
+	}
+}
+
+func addIntFlags(command *cobra.Command, metadata map[string]FlagMetadata[int]) {
+	for _, value := range metadata {
+		if value.IsInheritable {
+			var inheritFlagName = value.Name + "-inherit"
+			command.Flags().Int(value.Name, value.DefaultValue, value.Usage)
+			command.Flags().Bool(
+				inheritFlagName,
+				false,
+				fmt.Sprintf(inheritedUsageFormat, value.Name),
+			)
+		} else {
+			command.Flags().Int(value.Name, value.DefaultValue, value.Usage)
+		}
+	}
+}
+
+func addFloatFlags(command *cobra.Command, metadata map[string]FlagMetadata[float64]) {
+	for _, value := range metadata {
+		if value.IsInheritable {
+			var inheritFlagName = value.Name + "-inherit"
+			command.Flags().Float64(value.Name, value.DefaultValue, value.Usage)
+			command.Flags().Bool(
+				inheritFlagName,
+				false,
+				fmt.Sprintf(inheritedUsageFormat, value.Name),
+			)
+		} else {
+			command.Flags().Float64(value.Name, value.DefaultValue, value.Usage)
+		}
+	}
+}
+
+func addStringSliceFlags(command *cobra.Command, metadata map[string]FlagMetadata[[]string]) {
+	for _, value := range metadata {
+		if value.IsInheritable {
+			var inheritedFlagName = value.Name + "-inherit"
+			command.Flags().StringSlice(value.Name, value.DefaultValue, value.Usage)
+			command.Flags().Bool(
+				inheritedFlagName,
+				false,
+				fmt.Sprintf(inheritedUsageFormat, value.Name),
+			)
+			command.MarkFlagsMutuallyExclusive(value.Name, inheritedFlagName)
+		} else {
+			command.Flags().StringSlice(value.Name, value.DefaultValue, value.Usage)
+		}
+	}
+}
+
+func addMapFlags(command *cobra.Command, metadata map[string]FlagMetadata[map[string]string]) {
+	for _, value := range metadata {
+		if value.IsInheritable {
+			var inheritedFlagName = value.Name + "-inherit"
+			command.Flags().StringToString(value.Name, value.DefaultValue, value.Usage)
+			command.Flags().Bool(
+				inheritedFlagName,
+				false,
+				fmt.Sprintf("Mark %s as inherited and remove its concrete value", value.Name),
+			)
+			command.MarkFlagsMutuallyExclusive(value.Name, inheritedFlagName)
+		} else {
+			command.Flags().StringToString(value.Name, value.DefaultValue, value.Usage)
+		}
+	}
+}
+
+func addCommonArgs(command *cobra.Command) {
+	addStringFlags(command, commonStringFlagMetadata)
+	addStringSliceFlags(command, commonStringSliceFlagMetadata)
+}
+
+// RemoveItemRecursive accesses the given flags and attempts to remove a given item
+func RemoveItemRecursive(cmd *cobra.Command, args []string, what string) {
+	_ = args
```

Comment:
Yes, this is a noop so the compiler doesn't complain about unused arguments. We want to have this argument here because we might need to access this in the future. The variable contains another representation of the given arguments if I understand the cobra docstring correctly.